spec/requests/topics_controller_spec.rb

```diff
@@ -1792,6 +1792,34 @@ def fabricate_topic(user, category = nil)
     end
   end
 
+  describe '#show_by_external_id' do
+    fab!(:private_topic) { Fabricate(:private_message_topic, external_id: 'private') }
+    fab!(:topic) { Fabricate(:topic, external_id: 'asdf') }
+
+    it 'returns 301 when found' do
+      get "/t/external_id/asdf.json"
+      expect(response.status).to eq(301)
+      expect(response).to redirect_to(topic.relative_url + ".json?page=")
+    end
+
+    it 'returns right response when not found' do
+      get "/t/external_id/fdsa.json"
+      expect(response.status).to eq(403)
```

Comment:
```suggestion
      expect(response.status).to eq(404)
```

Not found should be 404 in this case.